Add Jumpstart status REST API endpoint

tyxla opened the pull request:
This PR introduces a simple endpoint for retrieving current Jumpstart status. Includes tests.

We will eventually use this endpoint in Calypso for enabling the Jumpstart functionality there. So in addition to being able to run the currently existing jumpstart endpoint, we'll need the current status in order to determine whether the Jumpstart functionality should be available or not. Already merged PR that relies on this endpoint can be found here: https://github.com/Automattic/wp-calypso/pull/9868. 

To test:

* Checkout this branch.
* Verify all tests pass ✅ .

nylen commented on _inc/lib/class.core-rest-api-endpoints.php:
```diff
@@ -216,7 +216,14 @@ public static function register_endpoints() {
+		register_rest_route( 'jetpack/v4', '/jumpstart', array(
+			'methods' => WP_REST_Server::READABLE,
+			'callback' => __CLASS__ . '::jumpstart_status',
+			'permission_callback' => __CLASS__ . '::update_settings_permission_check',
```
Suggest aligning this block on the `=>` using spaces - this is the pattern used throughout WP core, and it makes the code much easier to read.

nylen commented:
Even though this new endpoint is very simple, please add a unit test or two.

tyxla commented:
Thanks for the feedback @nylen! I've addressed your feedback, adding initial tests and improving array definitions. 

In any case this PR needs some more work and testing; it's meant to stay in progress at least until we merge #5418 and D3588-code.

Also, it's worth to mention that this one is useful for anyone working on https://github.com/Automattic/wp-calypso/pull/9868

georgestephanis commented:
It looks like with the merging in of some other changesets that some additional lines are showing in the PR that maybe shouldn't be?

tyxla commented:
@georgestephanis actually, this PR needs some initial commits from #5418 in order to work. That's why they're included in this branch. These commits are already squashed in #5418, so you probably won't find them though.

My plan is to rebase and remove all of those commits as soon as #5418 is merged. This is also the main reason why this PR is still marked as "In progress".

roccotripaldi commented:
Looks good! Thanks for adding tests.

I think we can merge this. Once #5418 lands, we can try sending remote requests to this endpoint, and make adjustments if need be.

jeherve commented:
Cherry-picked to `branch-4.5` too.
